library/core/src/ptr/alignment.rs

```diff
@@ -370,3 +376,74 @@ enum AlignmentEnum {
     _Align1Shl62 = 1 << 62,
     _Align1Shl63 = 1 << 63,
 }
+
+#[cfg(kani)]
+#[unstable(feature="kani", issue="none")]
+mod verify {
+    use super::*;
+
+    // pub const fn of<T>() -> Self
+    #[kani::proof]
+    pub fn check_of_i32() {
+        let alignment = Alignment::of::<i32>();
+        assert!(alignment.as_usize().is_power_of_two());
+    }
+
+    // pub const fn new(align: usize) -> Option<Self>
+    #[kani::proof_for_contract(Alignment::new)]
+    pub fn check_new() {
+        let a = kani::any::<usize>();
+        let alignment_opt = Alignment::new(a);
+        match alignment_opt {
+            Some(alignment) => assert_eq!(alignment.as_usize(), a),
+            None => assert!(!a.is_power_of_two())
+        }
```

Comment:
All of these assertions should just be in the ensures clause. If we can't write them in the ensures clause, then we should add a comment here saying why.

Reply:
I'll review to make sure they are all added as ensures clauses. So the harnesses then shouldn't have any asserts?

Reply:
Done. `check_as_usize` is an example where I think the property can't be expressed through a contract (on a single function), but I don't think that's a shortcoming of Kani's.